lib/trace_parallel.c

```diff
@@ -426,6 +426,13 @@ static libtrace_thread_t * get_thread_descriptor(libtrace_t *libtrace) {
 }
 
 DLLEXPORT void libtrace_make_packet_safe(libtrace_packet_t *pkt) {
+
+        // Can the format module do this beter than copying the
+        // entire packet?
+        if (pkt->trace && pkt->trace->format->safe_packet)
+            if (pkt->trace->format->safe_packet(pkt) > 0)
+                return;
```

Comment:
Actually, looking at this more, we need another function for this.

The purpose of libtrace_make_packet_safe() is to copy the packet out of memory allocated for the format, so the trace can be paused and the packets remain valid. Which requires the copy.

Holding a packet for later processing is a separate issue again.
Hooking this same logic into a new function libtrace_hold_packet() seems the correct thing to do here.